src/client/Microsoft.Identity.Client/TokenSource.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Specifies the source of the access and Id tokens in the authentication result.
+    /// </summary>
+    public enum TokenSource
+    {
+        /// <summary>
+        /// The source of access and Id token is not set.
+        /// </summary>
+        None = 0,
+        /// <summary>
+        /// The source of access and Id token is MSAL's cache.
+        /// </summary>
+        Cache,
+        /// <summary>
+        /// The source of the access and Id token is Identity Provider - Azure Active Directory (AAD), ADFS or AAD B2C.
+        /// </summary>
+        IdentityProvider,
+        /// <summary>
```

Comment:
nit: would you want to separate b2c from AAD and ADFS or does it matter? Should it be called `AuthorizationServer`? When i think of an Identity provider, I think of Google or FB or Microsoft...really just a question

Reply:
The goal of this small improvement is for developers, especially on Identity.Web, to monitor their token cache hit ratio. There are other proposals around providing more details such as "which broker (name and version) was used?", but will treat them separately. 

Not sure about the naming, I proposed the OAuth2 term. @jmprieur - what do you think?